controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -297,21 +297,23 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	kcp.Status.Replicas = replicas
 
 	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster), r.scheme)
-	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) {
+	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) && !apierrors.IsTimeout(errors.Cause(err)) {
 		return errors.Wrap(err, "failed to create remote cluster client")
 	}
 
 	readyMachines := int32(0)
```

Comment:
Please move this declaration up above the call to `r.removeClientGetter` so that the additional handling of `err` below is easier to correlate with the call to `r.removeClientGetter` above.

Reply:
fixed, thank you